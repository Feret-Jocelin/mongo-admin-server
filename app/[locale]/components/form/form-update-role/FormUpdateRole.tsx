'use client';
import { useEffect, useState } from 'react';
import { useTranslations } from 'next-intl';

import GenericModal from '@components/modal/generic/GenericModal';

import { useDispatch, useSelector } from '@/store/store';
import { updateRole, setErrorUser, selectError, selectUserSelected} from '@/domain/usecases/user-slice';
import SelectInput from '../../ui/inputs/select/SelectInput';
import { RoleType } from '@/domain/entities/user-types';

interface UpdateRoleModalProps {
  userName: string,
  open: boolean;
  onClose: () => void;
}

export default function FormUpdateRole({
  userName,
  open,
  onClose,
}: UpdateRoleModalProps) {
  /* Static Data */
  const dispatch = useDispatch();
  const t = useTranslations();
  const errorSelector = useSelector(selectError);

  const optionsRoles = [
    { value: 'read', label: t('role.read'), },
    { value: 'readWrite', label: t('role.readWrite'), },
    { value: 'dbAdmin', label: t('role.dbAdmin'), },
    { value: 'dbOwner', label: t('role.dbOwner'), },
    { value: 'userAdmin', label: t('role.userAdmin'), },
    { value: 'clusterAdmin', label: t('role.clusterAdmn')},
    { value: 'backup', label: t('role.backup')},
    { value: 'root', label: t('role.root')}
  ];
  /* Methods */
  const handleSubmit = async () => {}
  
  const handleRoleChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    const selectedValue = event.target.value;
    setRoles(selectedValue); 
  };
  /* Local Data */

  const [roles, setRoles] = useState<string>('');
  const [errorInput, setErrorInput] = useState<string>('');
  const errors = errorInput || errorSelector;

  useEffect(() => {
    dispatch(setErrorUser(''));
  }, [open, dispatch])
  
  return (
    <GenericModal
      open={open}
      title={t('user.update')}
      onClose={onClose}
      confirmLabelTrad={t('modal.button.update')}
      typeButton='submit'
      onValidate={handleSubmit}
    >
      <form>
        <SelectInput
          label={t('user.role')}
          options={optionsRoles}
          value={roles} 
          onChange={handleRoleChange}
        />
      </form>
    </GenericModal>
  );
}
