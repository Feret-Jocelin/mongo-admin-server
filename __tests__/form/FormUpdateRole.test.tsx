/* eslint-disable no-undef */
import React from 'react';
import { render, screen } from '@testing-library/react';
import { NextIntlClientProvider } from 'next-intl';
import { store } from '@/store/store';
import { Provider } from 'react-redux';
import FormUpdateRole from '@/app/[locale]/components/form/form-update-role/FormUpdateRole';

jest.mock('next-intl/client', () => ({
  usePathname() {
    return '';
  },
  useRouter() {
    return {
      push: () => jest.fn(),
      replace: () => jest.fn(),
    };
  },
}));

const locale = 'fr';
const message = require(`../../shared/locales/${locale}.json`)
const mockUser = {
  "userId": "b4c2885f-c976-4f1e-8de1-0481659eb5ce",
  "user": "itachiUchiwa",
  "db": "localDbUser",
  "roles": [
    {
      "role": "readWrite",
      "db": "localDbUser"
    }
  ]
}

describe("FormUpdateRole", () => {
  it('renders', () => {
    const handleClose = jest.fn();
    const { container } = render(
      <Provider store={store}>
        <NextIntlClientProvider locale={locale} messages={message}>
          <FormUpdateRole
            userName={mockUser}
            open={true}
            onClose={handleClose}
          />
        </NextIntlClientProvider>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  })
})