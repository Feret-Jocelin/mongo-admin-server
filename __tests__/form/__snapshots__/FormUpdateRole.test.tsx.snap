// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormUpdateRole renders 1`] = `
<div>
  <dialog
    class="modal"
  >
    <section
      class="modal__header"
    >
      <div
        class="modal__header--icon"
      >
        <div
          class="undefined pointer-cursor"
        />
      </div>
      <h2>
        Modifier un rôle
      </h2>
    </section>
    <section
      class="modal__container"
    >
      <form
        class="form-update-role"
      >
        <div
          class="input-container"
        >
          <label
            class="label"
          >
            user.database
          </label>
          <input
            class="input "
            placeholder="user.database"
            type="text"
            value="localDbUser"
          />
        </div>
        <div
          class="container"
        >
          <label>
            Rôles
          </label>
          <select
            class="select "
          >
            <option
              disabled=""
              value=""
            >
              Choisir
            </option>
            <option
              value="read"
            >
              Lecture seule
            </option>
            <option
              value="readWrite"
            >
              Lecture et écriture
            </option>
            <option
              value="dbAdmin"
            >
              Administrateur de base de données
            </option>
            <option
              value="dbOwner"
            >
              Propriétaire de la base de données
            </option>
            <option
              value="userAdmin"
            >
              Administrateur d'utilisateur
            </option>
            <option
              value="clusterAdmin"
            >
              Administrateur de cluster
            </option>
            <option
              value="backup"
            >
              Sauvegarde
            </option>
            <option
              value="root"
            >
              Superutilisateur
            </option>
          </select>
          <div
            class="iconContainer"
          >
            <div
              class="undefined "
            />
          </div>
        </div>
      </form>
    </section>
    <section
      class="modal__footer"
    >
      <button
        class="button full outline center"
        type="button"
      >
        Annuler
      </button>
      <button
        class="button full center"
        type="submit"
      >
        Modifier
      </button>
    </section>
  </dialog>
</div>
`;
